info.limpet.test/src/info/limpet/data/TestOperations.java

```diff
@@ -800,6 +803,39 @@ public void testSubtraction()
   }
 
   @SuppressWarnings("unchecked")
+  public void testAddLayerOperation()
+  {
+    // place to store results data
+    InMemoryStore store = new SampleData().getData(10);
+
+    List<IStoreItem> selection = new ArrayList<IStoreItem>();
+
+    StoreGroup track1 = new StoreGroup("Track 1");
+    selection.add(track1);
+
+    Collection<ICommand<IStoreItem>> commands =
+        new AddLayerOperation().actionsFor(selection, store, context);
+    assertEquals("invalid number of inputs", 1, commands.size());
```

Comment:
hmm, I think the message for this test is that it is proving the "valid number of commands", not an invalid number of inputs.